connect/runtime/src/main/java/org/apache/kafka/connect/tools/SchemaSourceTask.java

```diff
@@ -160,14 +161,22 @@ public void start(Map<String, String> props) {
             return result;
         } else {
             synchronized (this) {
-                this.wait();
```

Comment:
Looking a little bit closer what `throttler` does, it seems that it uses its own condition variable internally for throttling. That explains the call to `throttler.wakeup()` below (that was expected to wake up the throttler). The problem (bug) of course is that this call to wait() here is not waiting on the same object as the throttler (one object is the `SchemaSourceTask` and the other one is the `ThroughputThrottler`). 

There are two alternative solutions to introducing a new condition (here `stopped`).

1. Just keep on calling throttler.throttle(): 
  - if target throughput is <= 0, it will practically wait indefinitely until a thread calls `stop` -> `throttler.wakeup()`. 
  - if target throughput is > 0, is will still wait for approximately 1/throughput time. 
2. Since you can't reset the target throughput for a specific throttler (it's a constructor argument), discard the old throttler and create a new one with target throughput 0. Here that'd be: 
```java
throttler = new ThroughputThrottler(0, 0)
```
 (the 2nd argument doesn't matter if the 1st one is 0). This new instantiation would still need to be wrapped with `synchronized (this)`.

Now, all that needs to happen is to wrap `throttler.wakeup();` within a `synchronized (this)` block, in order to make visible to every thread the switch between throttlers. 

The implementation is far smaller than the explanation (3 lines if I can count well). Your call.